File: parsedx/parsedx.py
def parsedx(filename) :
    """PARSEDX
    ---------------------------------------------------------------------------
    DESCRIPTION
    Parsedx accepts an Excel file containing strings recorded from a Dillon 
    EDXtreme dynamometer and returns data parsed into separate datetime, 
    elapsed seconds, instantaneous force, and peak force columns. The strings
    are printed by the Dillon EDXtreme in continuous recording mode in #4 
    format with the dynamometer connected to a computer by a serial port 
    connection and WedgeLink software. 
    
    INPUTS
    filename: string - filepath of Excel file containing original strings
    
    OUTPUTS
    B: DataFrame - m x 4 DataFrame containing datetime, elapsed seconds,
    instantaneous force, and peak force records in separate columns
    ---------------------------------------------------------------------------    
    """
    import pandas as pd
    import re
    import matplotlib.pyplot as plt
    df=pd.read_excel(filename,sheet_name=0,header=None)
    df.columns = ['strings']
    dtstrings = [re.search('\d*\s\w*\s\d{4}[\,]\d*\:\d{2}\:\d{2}',string) for string in df.strings]
    dt = [pd.to_datetime(dtstring.group(),format='%d %b %Y,%H:%M:%S') for dtstring in dtstrings]
    t = [(dts - dt[0]).total_seconds() for dts in dt]
    fstrings = [re.findall('\d*\.\d{1}',string) for string in df.strings]
    f_ins = []
    f_pk = []
    for b in fstrings :
        f_ins.append(pd.to_numeric(b[0]))
        f_pk.append(pd.to_numeric(b[1]))
    A = pd.DataFrame([dt,t,f_ins,f_pk]).transpose()
    A.columns = ['Datetime','Seconds','F_ins','F_pk']
    A.to_excel(filename,sheetname=2,index=0)
    return A
    A['F_ins'].plot(x='Seconds')
    plt.ylabel('Force, F (kg)')
    plt.xlabel('Time, t (sec)')
    plt.show()

File: parsedx/parsedx.m
function [C,fig_handle] = parsedx(filename)
%{
PARSEDX
---------------------------------------------------------------------------
DESCRIPTION
Parsedx accepts an Excel file containing strings recorded from a Dillon 
EDXtreme dynamometer and returns data parsed into separate datetime, 
elapsed seconds, instantaneous force, and peak force columns. The strings
are printed by the Dillon EDXtreme in continuous recording mode in #4 
format with the dynamometer connected to a computer by a serial port 
connection and WedgeLink software. 

INPUTS
filename: string - filepath of Excel file containing original strings

OUTPUTS
B: matrix - m x 4 matrix containing datetime, elapsed seconds,
instantaneous force, and peak force records in separate columns
fig_handle: figure object - line plot of elapsed seconds and instantaneous
force
---------------------------------------------------------------------------
%}
% Read file
[~,A,~]=xlsread(filename);
% Initialize array
B = zeros(size(A,1),2);
% Read and convert datetime stamps
dt = regexp(A,'\d*\s\w*\s\d{4}[\,]\d*\:\d{2}\:\d{2}','match');
t = datetime(string(dt),'InputFormat','d MMM y,H:mm:ss');
s = seconds(t-t(1));
% Read and convert force measurements
f = regexp(A,'\d*\.\d{1}','match');
for i = 1:size(f,1)
    B(i,1:2) = [str2double(f{i,1}{1,1}) str2double(f{i,1}{1,2})];
end
C = table(t,s,B(:,1),B(:,2),'VariableNames',{'Datetime','Seconds',...
    'F_ins','F_pk'});
% Plot
fig_handle = plot(s,B(:,1));
xlabel('Time, t (sec)');
ylabel('Force, F (kg)');
writetable(C,filename,'sheet',2);